pkg/serviceprovider/github/downloadfilecapability.go

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+
+	"github.com/google/go-github/v45/github"
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/serviceprovider"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type downloadFileCapability struct {
+	httpClient      *http.Client
+	ghClientBuilder githubClientBuilder
+}
+
+var _ serviceprovider.DownloadFileCapability = (*downloadFileCapability)(nil)
+
+var (
+	unexpectedStatusCodeError  = errors.New("unexpected status code from GitHub API")
+	fileSizeLimitExceededError = errors.New("failed to retrieve file: size too big")
+	pathIsADirectoryError      = errors.New("provided path refers to a directory, not file")
+)
+
+var _URLRegexp = regexp.MustCompile(`(?Um)^(?:https)(?:\:\/\/)github.com/(?P<owner>[^/]+)/(?P<repo>[^/]+)(.git)?$`)
+var _URLRegexpNames = _URLRegexp.SubexpNames()
+var maxFileSizeLimit int = 2097152
```

Comment:
How did we choose 2MB? Could this be made configurable?

Reply:
\+